### PR TITLE
fix(dialog): use CompositionTransaction in dialog

### LINKED ISSUE #145: dialog with composed body not centering vertically
There is a problem with displaying dialog content when it composed into body
```javascript
...
        <ai-dialog-body>
            <compose view-model.bind="bodyViewModel" model.bind="bodyModel"></compose>
        </ai-dialog-body>
...
```
The modal opens centered vertically as if there is no content and expanded down with composed content.

### PR DESCRIPTION
**Pull request by jedd-ahyoung**
Use CompositionTransaction in dialog to improve rendering.

Fixes #145.

Signed-off-by: Jedd Ahyoung <jedd.ahyoung@gmail.com>

### CODE REVIEW

**Comment by jedd-ahyoung** [comment]
@EisenbergEffect I had some trouble getting the tests to run for this, but the sample was working correctly. I think this might be due to dependencies but I'm not entirely sure. If there's a way to test this with release, let me know.

File: src/dialog-service.js [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 import {Container} from 'aurelia-dependency-injection';
-import {CompositionEngine, ViewSlot} from 'aurelia-templating';
+import {CompositionEngine, CompositionTransaction, ViewSlot} from 'aurelia-templating';
 import {DialogController} from './dialog-controller';
 import {Renderer} from './renderers/renderer';
 import {invokeLifecycle} from './lifecycle';
 
+console.log("CompositionTransaction", CompositionTransaction);
+console.log("CompositionEngine", CompositionEngine);
+console.log("Container", Container);
```
Comment (EisenbergEffect) [inline src/dialog-service.js]: Can you remove these please?

Comment (jedd-ahyoung) [inline src/dialog-service.js]: Whoops. Yes.

**Comment by EisenbergEffect** [comment]
@jedd-ahyoung Are you able to fix and merge soon? If so, I can get this in today's release. Needs to happen in the next 30 minutes though.

**Comment by jedd-ahyoung** [comment]
I can't get this in in the next 30 minutes, unfortunately. I took a look
this morning and identified another issue that needs to be fixed. I may be
able to get this in today, though.

2016-05-24 11:27 GMT-04:00 Rob Eisenberg <notifications@github.com>:

**Comment by EisenbergEffect** [comment]
@jedd-ahyoung What is the status on this? I'd like to try to get it into tomorrow's release?

**Comment by EisenbergEffect** [comment]
Status please.

**Pull request closed without merging**